cmd/derper/derper.go

```diff
@@ -241,6 +242,12 @@ func main() {
 		ReadTimeout:  30 * time.Second,
 		WriteTimeout: 30 * time.Second,
 	}
+	go func() {
+		<-ctx.Done()
+		timeoutCtx, cancel := context.WithTimeout(context.Background(), time.Second)
```

Comment:
I might be missing something, but at the moment this timeout does not seem to achieve much, since `Shutdown` will make `httpsrv.Serve` return immediately, which will return from `main`. This is probably fine, but perhaps I'd suggest to leave a comment calling this out explicitly.

Reply:
Shutdown waits until client connections / channels become idle before disconnecting clients. This would wait indefinitely unless it is passed a timeout. See https://pkg.go.dev/net/http#Server.Shutdown

Reply:
Yes, when I mentioned that `Serve` will return immediately and we'll terminate the process, I was referring to the second paragraph of the godoc you linked to:

> When Shutdown is called, Serve, ListenAndServe, and ListenAndServeTLS immediately return ErrServerClosed. Make sure the program doesn't exit and waits instead for Shutdown to return.

So it seems that without something like a sync.WaitGroup, we're likely to have the process terminate before `Shutdown` gets to do any of the waiting/closing.

Reply:
I see, you're absolutely right. I've removed the timeout, immediate termination is fine here.